.github/workflows/ci.yml

```
@@ -11,25 +11,68 @@ on:
     branches:
       - "**:**"
 
+env:
+  DOCKER_COMPOSE: docker-compose -f docker-compose.yml -f docker-compose.ci.yml
+
 jobs:
-  test:
+  setup:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+      - uses: ./.github/actions/docker-cache
+      - uses: ./.github/actions/gems-cache
 
-      - name: Set environment up
+      - name: Install dependencies
         run: |
-          docker-compose build
-          docker-compose run --rm -e RAILS_ENV=test web bundle install
-          docker-compose run --rm -e RAILS_ENV=test web bundle exec rake db:setup db:test:prepare elasticsearch:setup
-          docker-compose up -d db elasticsearch redis sshd csv_watch ftp_monitor sidekiq
+          $DOCKER_COMPOSE run --rm --no-deps -e RAILS_ENV=test web bundle install --jobs 4
+          $DOCKER_COMPOSE run --rm --no-deps -e RAILS_ENV=test web bundle clean --force
+
+  unit_tests:
+    runs-on: ubuntu-latest
+    needs: setup
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/docker-cache
+      - uses: ./.github/actions/gems-cache
+
+      - name: Setup databases
+        run: |
+          $DOCKER_COMPOSE up -d db elasticsearch redis
+          $DOCKER_COMPOSE run --rm -e RAILS_ENV=test web bundle exec rake db:setup db:test:prepare elasticsearch:setup
+
+      - name: Start services
+        run: $DOCKER_COMPOSE up -d sshd csv_watch ftp_monitor sidekiq
 
       - name: Run specs
-        run: docker-compose run --rm -e RAILS_ENV=test web bundle exec rake
+        run: $DOCKER_COMPOSE run --rm -e RAILS_ENV=test web bundle exec rspec
+
+  integration_tests:
+    runs-on: ubuntu-latest
+    needs: setup
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/docker-cache
+      - uses: ./.github/actions/gems-cache
+
+      - name: Setup databases
+        run: |
+          $DOCKER_COMPOSE up -d db elasticsearch redis
+          $DOCKER_COMPOSE run --rm -e RAILS_ENV=test web bundle exec rake db:setup db:test:prepare elasticsearch:setup
+
+      - name: Start services
+        run: $DOCKER_COMPOSE up -d sshd csv_watch ftp_monitor sidekiq
+
+      - name: Run integration specs (rspec + capybara)
+        run: $DOCKER_COMPOSE run --rm -e RAILS_ENV=test web bundle exec rspec spec/features/*_spec.rb
```

Why the file-level wildecard? If we'd start organizing feature specs in folders, they wouldn't be matched.
```suggestion
        run: $DOCKER_COMPOSE run --rm -e RAILS_ENV=test web bundle exec rspec spec/features/
```

---

This is to work around the exclude-pattern configuration!

Which we should remove, yes.

---

Yes, please let's do that instead 🙏